solver/pb/ops.proto

```diff
@@ -211,4 +199,67 @@ message Definition {
 message HostIP {
 	string Host = 1;
 	string IP = 2;
+}
+
+message FileOp {
+	repeated FileAction actions = 2;
+}
+
+message FileAction {
+	int64 input = 1 [(gogoproto.customtype) = "InputIndex", (gogoproto.nullable) = false]; // could be real input or target (target index + max input index)
+	int64 secondaryInput = 2 [(gogoproto.customtype) = "InputIndex", (gogoproto.nullable) = false]; // --//--
+  int64 output = 3 [(gogoproto.customtype) = "OutputIndex", (gogoproto.nullable) = false];
+	oneof action {
+		FileActionCopy copy = 4;
+		FileActionMkFile mkfile = 5;
+		FileActionMkDir mkdir = 6;
+		FileActionRm rm = 7;
+	}
+}
+
+message FileActionCopy {
+	string src = 1;
+	string dest = 2;
+	ChownOpt owner = 4;
+	int32 mode = 5;
+	bool followSymlink = 6;
+	bool dirCopyContents = 7;
+	bool attemptUnpack = 8;
```

Comment:
what are the forward/backward compatibility implications here? Adding a new archive type that could be unpacked can't be done I think since it would change the semantics. `s/attempt/must/g` might be a simple way to avoid that?

Reply:
This is actually a very bad feature and only here because of the dockerfile compatibility requirements. Normally, extraction is too high-level for LLB and should be done with an extra op. I think I'll rename it `attemptUnpackDockerCompatibility` to make it clear that new developments will not be accepted. If the behavior was `mustUnpack` then it would be very easy to prepend the input with `exec` and we wouldn't need it.